.circleci/config.yml

```diff
@@ -173,30 +186,111 @@ jobs:
           command : bash CircleciScripts/check_test_result.sh  "test_result"         
       - store_artifacts:
           path: test_result
+  pre_integrationtest:
+    macos: 
+      xcode: "10.1.0"
+    steps:
+      - run:
+          name: check environment variabe 
+          command: |
+              if [ "${SKIP_INTEGRATIONTEST}" == "TRUE" ]; then
+                echo "project enviroment variabe SKIP_INTEGRATIONTEST should not be TRUE at the moment, this will skip all integration tests"
+                exit 1
+              fi    
+      - checkout
+      - set_enviroment_variables
+      - configure_aws
+      - run:
+          name: create credentials
+          command: |
+            echo "${destination}" 
+            aws s3 cp s3://ios-circleci-payload/credentials.json  AWSCoreTests/Resources/credentials.json 
+            aws s3 cp s3://ios-circleci-payload/awsconfiguration.json  AWSAuthSDK/Tests/AWSMobileClientTests/awsconfiguration.json
+            aws s3 cp s3://ios-circleci-payload/credentials-mc.json   AWSAuthSDK/Tests/AWSMobileClientTests/credentials-mc.json     
+      - pre_start_simulator   
+      - run:
+          name: build sdk
+          command: |
+            xcodebuild  build-for-testing -project AWSiOSSDKv2.xcodeproj -scheme AWSAllTests -sdk iphonesimulator -destination  "${destination}"   -derivedDataPath  ./build_result/AWSiOSSDKv2        
+            xcodebuild  build-for-testing -project AWSAuthSDK/AWSAuthSDK.xcodeproj -scheme AWSMobileClient -sdk iphonesimulator -destination "${destination}"   -derivedDataPath  ./build_result/AWSAuthSDK  
+      - save_cache:
+          key: build_result-{{ .Revision }}
+          paths:
+            - build_result
+  post_integrationtest:
+    macos: 
+      xcode: "10.1.0"
+    steps:
+      #TODO need check why multiple key does not work here
+      - restore_cache:
+          key: test_result-{{ .Revision }}-DynamoDB
+      - restore_cache:
+          key: test_result-{{ .Revision }}-Polly          
+      - restore_cache:
+          key: test_result-{{ .Revision }}-S3   
+      - restore_cache:
+          key: test_result-{{ .Revision }}-Pinpoint
+      - restore_cache:
+          key: test_result-{{ .Revision }}-IoT          
+      - restore_cache:
+          key: test_result-{{ .Revision }}-KMS_ELB_EC2_Lambda_Lex                      
```

Comment:
can you expand the names? Abbreviation makes it hard to understand what tests fail?

Reply:
done